### DISCUSSION TITLE
podman checkpoint Error: configured runtime does not support checkpoint/restore

### DISCUSSION THREAD

**Discussion by chengxilo** [containers/podman#22123] [Q&A] [answered]
Here is what I've tried in my test environment.
```
root@iZ2zef7sqxs9idaeluuxe0Z:~# podman container ps
CONTAINER ID  IMAGE                           COMMAND           CREATED         STATUS             PORTS                 NAMES
0069bb9f997e  docker.io/library/httpd:latest  httpd-foreground  25 minutes ago  Up 25 minutes ago  0.0.0.0:8080->80/tcp  brave_villani
root@iZ2zef7sqxs9idaeluuxe0Z:~# podman container checkpoint -l
Error: configured runtime does not support checkpoint/restore
```
I want to ask for what I can do to solve this problem and why it happening.

# my podman version 
```
root@iZ2zef7sqxs9idaeluuxe0Z:~# podman version
Version:      3.4.4
API Version:  3.4.4
Go Version:   go1.18.1
Built:        Thu Jan  1 08:00:00 1970
OS/Arch:      linux/amd64
```

# my criu version
```
root@iZ2zef7sqxs9idaeluuxe0Z:~# criu --version
Version: 3.16.1
```

**Comment by edsantiago** [comment, marked as answer]
Please try:
```console
# podman info --format '{{.Host.OCIRuntime.Path}}'
/usr/bin/crun    <--- YMMV
# /usr/bin/crun checkpoint --help    <----- Use the path that *your* podman shows
```
If this emits an error, then file a bug with your distro asking for checkpoint support in the runtime tool.

If this emits help-message-looking output, please post the full output of `podman info` here.

**Reply by chengxilo** [reply]
Thanks for your reply. And this is what I got.
```
root@iZ2zef7sqxs9idaeluuxe0Z:/# podman info --format '{{.Host.OCIRuntime.Path}}'
/usr/bin/crun
root@iZ2zef7sqxs9idaeluuxe0Z:/# /usr/bin/crun checkpoint --help 
2024-03-21T18:27:35.000661979Z: unknown command checkpoint
root@iZ2zef7sqxs9idaeluuxe0Z:/# 
```
It seems to be the first case?🥲

**Reply by edsantiago** [reply]
Afraid so. Please follow up with your distribution. It might be as simple as adding `criu` to the package build dependencies, but there is really nothing we (upstream) can do about this. Good luck!

**Reply by chengxilo** [reply]
OK. Thank you for your time and effort. 🫡

**Reply by achekunov** [reply]
Just try to switch the runtime to runc.
Helped on Ubuntu 24.04

**Comment by nnWhisperer** [comment]
Hello, I had the same problem; [downloading](https://labs.iximiuz.com/courses/containerd-cli/components/installation#oci-runtimes) [the runc](https://github.com/opencontainers/runc/releases/) and configuring podman to use it(by changing the `/usr/share/containers/containers.conf` file) fixed it. It didn't checkpoint an existing container since the container was still using the previous runtime. It worked in the next ones started after the runtime change.

@edsantiago thank you for the `/usr/bin/crun checkpoint --help    <----- Use the path that *your* podman shows` tip.
I wrote these to help future searchers having a similar problem coming here.
